wagtail/tests/test_avatar_url.py

```diff
@@ -0,0 +1,16 @@
+from django.test import TestCase
+from django.contrib.auth.models import User
+from wagtail.admin.templatetags.wagtailadmin_tags import avatar_url
+
+class AvatarUrlTest(TestCase):
+    def setUp(self):
+        self.user_with_custom_avatar = User.objects.create(username="fred_user")
+        self.user_with_default_avatar = User.objects.create(username="jane_user")
+
+    def test_custom_avatar_url(self):
+        url = avatar_url(self.user_with_custom_avatar, size=100)
```

Comment:
Instead, you can do something like:

```suggestion
        def custom_avatar_url(user, size):
            if "fred" in user.username:
                return f"https://example.com/avatars/fred-{size}.png"
            return None

        with self.register_hook("get_avatar_url", custom_avatar_url):
            url = avatar_url(self.user_with_custom_avatar, size=100)
```

Also could you please move the test to https://github.com/wagtail/wagtail/blob/9889bddcb8823369f7a26c6b6f4a3eb6784d5985/wagtail/admin/tests/test_templatetags.py#L34 instead?

Reply:
Can you  assign me this work ?

Reply:
What do you mean? 🤔
We don't assign issues or PRs to non-core team members, but you are free to work on any issue/PR. You can make these changes yourself and push to the same branch, and the PR will be updated automatically.

Reply:
okay sorry about that i didn't knew about this